docs/content/preview/contribute/core-database/build-from-src-ubuntu.md

```diff
@@ -51,48 +51,117 @@ AlmaLinux 8 is the recommended Linux development platform for YugabyteDB.
 
 {{< /note >}}
 
+The following instructions are for Ubuntu 20.04 or Ubuntu 22.04.
+
 ## Install necessary packages
 
 Update packages on your system, install development tools and additional packages:
 
 ```sh
-sudo apt-get update
+sudo apt update
+sudo apt upgrade -y
 packages=(
   autoconf
-  cmake
+  build-essential
   curl
+  gettext
   git
-  git
-  git
-  libtool
-  libtool
   locales
   ninja-build
-  patchelf
   pkg-config
-  python3-pip
-  python3-venv
   rsync
-  zip
 )
-sudo apt-get install -y "${packages[@]}"
+sudo apt install -y "${packages[@]}"
 sudo locale-gen en_US.UTF-8
 ```
 
+### Python 3
+
+{{% readfile "includes/python.md" %}}
+
+```sh
+sudo apt install -y libffi-dev python3 python3-dev python3-venv
+```
+
+### CMake 3
+
+{{% readfile "includes/cmake.md" %}}
+
+The CMake version in the package manager is too old (3.16), so manually download a release.
+
+```sh
+mkdir ~/tools
+curl -L "https://github.com/Kitware/CMake/releases/download/v3.25.2/cmake-3.25.2-linux-x86_64.tar.gz" | tar xzC ~/tools
+# Also add the following line to your .bashrc or equivalent.
+export PATH="$HOME/tools/cmake-3.25.2-linux-x86_64/bin:$PATH"
+```
+
+### /opt/yb-build
+
+{{% readfile "includes/opt-yb-build.md" %}}
+
+### Ninja (optional)
+
+Use [Ninja][ninja] for faster builds.
+
+```sh
+sudo apt install -y ninja-build
+```
+
+[ninja]: https://ninja-build.org
+
+### Ccache (optional)
+
+Use [Ccache][ccache] for faster builds.
```

Comment:
```suggestion
Install [Ccache][ccache] for faster builds as follows:
```

Reply:
This can be done for all platforms in a followup.